### ISSUE TITLE
Add new loading indicator to pattern library

### ISSUE THREAD

**Issue by newleeland** [silverstripe/silverstripe-admin#797]
### Potential AC:
As SilverStripe we want to replace SilverStripe branded loading indicator with a more generic one as (basically we're sub consciously anchoring into users of the CMS, slow loading is because its SilverStripe). 

### Potential design:
https://projects.invisionapp.com/dsm/silver-stripe/silver-stripe/folder/components/5b9eeec5e2cb730017f84a1f

Animations should look something like this:
https://codepen.io/newleeland/pen/VqBmyV

**Comment by maxime-rainville** [comment]
`<sarcarsm>` What about a Wordpress/Drupal branded loading indicator instead? `</sacarsm>`

**Comment by robbieaverill** [comment]
For reference, Bootstrap 4.2 was released a month ago which has a bunch of spinner components: https://getbootstrap.com/docs/4.2/components/spinners/. It _might_ make sense to make whichever design we choose to implement work as a variation or state of this component so it can use some of the OOB spinner component styles etc.

Matching Reactstrap update is [version 7 which supports this component](https://github.com/reactstrap/reactstrap/issues/1347).

**Comment by newleeland** [comment]
> For reference, Bootstrap 4.2 was released a month ago which has a bunch of spinner components: https://getbootstrap.com/docs/4.2/components/spinners/. It _might_ make sense to make whichever design we choose to implement work as a variation or state of this component so it can use some of the OOB spinner component styles etc.

Yeah bootstrap 4 spinners is perfect. We could just change the colour to our colours. be good to not add too many custom styling from our end! @robbieaverill
